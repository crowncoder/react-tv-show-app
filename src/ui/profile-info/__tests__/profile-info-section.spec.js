import React from 'react';
import Enzyme, { shallow } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import ProfileInfoSection from '../profile-info-section'
import ProfileInfoCoverImg from '../profile-info-cover-img'
import ProfileInfoDescription from '../profile-info-description'

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<ProfileInfo /> unit test', () => {
  it('renders correctly', () => {
    const sectionInfo = { image: { medium: 'abc.com/image' }, summary: 'a simple summary' }
    const wrapper = shallow(<ProfileInfoSection sectionInfo={sectionInfo} />);
    expect(wrapper.find(ProfileInfoCoverImg)).toHaveLength(1);
    expect(wrapper.find(ProfileInfoDescription)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});