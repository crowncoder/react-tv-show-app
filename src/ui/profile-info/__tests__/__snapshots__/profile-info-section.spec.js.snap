// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileInfo /> unit test renders correctly 1`] = `
<div
  className="profileInfo__section"
>
  <aside>
    <Memo()
      image={
        Object {
          "medium": "abc.com/image",
        }
      }
    />
  </aside>
  <article>
    <Memo()
      description="a simple summary"
    />
  </article>
</div>
`;
